ibis/backends/base/sql/compiler/query_builder.py

```diff
@@ -73,15 +73,26 @@ def _quote_identifier(self, name):
 
     def _format_in_memory_table(self, op):
         names = op.schema.names
-        raw_rows = (
-            ", ".join(
-                f"{val!r} AS {self._quote_identifier(name)}"
-                for val, name in zip(row, names)
+        if self.context.compiler.support_values_syntax_in_select:
+            raw_rows = (
+                ", ".join(
+                    f"{val!r} AS {self._quote_identifier(name)}"
+                    for val, name in zip(row, names)
+                )
+                for row in op.data.to_frame().itertuples(index=False)
             )
-            for row in op.data.to_frame().itertuples(index=False)
-        )
-        rows = ", ".join(f"({raw_row})" for raw_row in raw_rows)
-        return f"(VALUES {rows})"
+            rows = ", ".join(f"({raw_row})" for raw_row in raw_rows)
+            return f"(VALUES {rows})"
+        else:
+            raw_rows = (
+                ", ".join(
+                    f"{val!r} AS {self._quote_identifier(name)}"
+                    for val, name in zip(row, names)
+                )
+                for row in op.data.to_frame().itertuples(index=False)
+            )
```

Comment:
It looks like these two branches share all of the code up to `rows` variable definition. Is there a way we can avoid the duplication?

Reply:
Fixed.